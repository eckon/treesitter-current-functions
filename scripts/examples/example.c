#include <stdio.h>

#ifdef COMPILER_OPTION
void foo() {}
#endif

#if MAGIC_NUMBER > 1
void bar() {}
#else
void baz() {}
#endif

struct student
{
   char name[50];
   int height;
};

int main(){
}

bool8 sub_8039880(void)
{
  return (CountMailType(WONDER_MAIL_TYPE_SOS) != 0 || CountMailType(WONDER_MAIL_TYPE_OKD) != 0 || sub_8011C1C() == 2);
}
