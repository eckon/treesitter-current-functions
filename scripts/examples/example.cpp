#include "app.h"

#include <chrono>
#include <thread>

#include "../data/api.h"

#define COLOR_HIGHLIGHT 1
#define COLOR_MODAL_BORDER 2

App *App::instance = 0;

App *App::getInstance() {
  if (instance == 0) {
    instance = new App();
  }
  return instance;
}

App::~App() { endwin(); }

App::App() {}

int App::getKeyPress() {}

void App::pushKey(int key) {}

void App::drawMainWinList() {}
