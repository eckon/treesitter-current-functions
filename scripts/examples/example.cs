namespace Example.Model
{
    public class ExampleManager
    {
        public ExampleManager()
        {
        }

        public async Task<foo> Start(int number, string test)
        {
        }

        private string End()
        {
        }
    }
}
