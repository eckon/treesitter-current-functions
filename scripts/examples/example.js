function multiline(
  a, b
) {}

function inline() {}

function inline2() {}

function inline3() {}

const arrowFunction = () => {};

class FirstClass {
  constructor(a) {}

  firstMethod() {}

  secoundMethod(a, v, c) {}
}

export class SecoundClass {
  thirdMethod() {}
}
