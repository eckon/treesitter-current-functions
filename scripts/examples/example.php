<?php

function firstFunction() {}

function secoundFunction() {}

class FirstClass {
    function firstMethod() {}
}
