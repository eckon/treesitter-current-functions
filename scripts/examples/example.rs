
#[allow(dead_code)]
struct Test1 {
    foo: usize
}
impl Test1 {
    fn new() -> Self {
        Test1 { foo: 0 }
    }
}

#[allow(dead_code)]
enum Test2 {
    Item1,
    Item2,
    Item3,
}
impl Test2 {
    fn index(&self) -> usize {
        match self {
            Test2::Item1 => 0,
            Test2::Item2 => 1,
            Test2::Item3 => 2,
        }
    }
}

#[allow(unused_variables)]
fn main() {
    let test1 = Test1::new();
    let test2 = Test2::Item1;
    test2.index();
}
