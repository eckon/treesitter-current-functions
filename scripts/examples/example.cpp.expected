{ {
    function_name = "getInstance",
    line_number = 13
  }, {
    function_name = "~App",
    line_number = 20
  }, {
    function_name = "App",
    line_number = 22
  }, {
    function_name = "getKeyPress",
    line_number = 24
  }, {
    function_name = "pushKey",
    line_number = 26
  }, {
    function_name = "drawMainWinList",
    line_number = 28
  } }